### PR TITLE
refactor(esl-utils): rework public api for SyntheticEventTarget

### PR DESCRIPTION
**Pull request by fshovchko**
In the scope of this PR I reworked public API for SyntheticEventTarget so future changes to the class won't be affecting other components

### CODE REVIEW

**Comment by ala-n** [comment]
Basically, I need to ask, @exadel-inc/esl-core-team, 
As you see in the current PR we going to share an ability to get a whole list of observers from EventTarget instance,
but the native implementation deprecates any possibility to access listeners outside debug.
And honestly, I partially see it's reasonable to have subscriptions be encapsulated and hidden from the outside world. 
Just an option we can think about the method to access just the whole state of Target instance, something like
```ts
   this.hasListeners('change'); // or similar short name
```

**Comment by fshovchko** [comment]
> Basically, I need to ask, @exadel-inc/esl-core-team, As you see in the current PR we going to share an ability to get a whole list of observers from EventTarget instance, but the native implementation deprecates any possibility to access listeners outside debug. And honestly, I partially see it's reasonable to have subscriptions be encapsulated and hidden from the outside world.
> 
> Just an option we can think about the method to access just the whole state of Target instance, something like
> 
> ```ts
>    this.hasListeners('change'); // or similar short name
> ```
> 
> From the usage point of view as you can see from the occurrence in the current PR it's more than enough to cover needs of `SyntheticEventTarget`
> 
> So, in case you prefer to enclose observers and go with the conservative `has` method instead of `get` mark this comment with the like +1emojy

this.hasListeners won't brake any of the code that is using SyntheticEventTarget api so I agree to change the method

**Review by NastaLeo — APPROVED** [review]

**Comment by ala-n** [comment]
> > Basically, I need to ask, @exadel-inc/esl-core-team, As you see in the current PR we going to share an ability to get a whole list of observers from EventTarget instance, but the native implementation deprecates any possibility to access listeners outside debug. And honestly, I partially see it's reasonable to have subscriptions be encapsulated and hidden from the outside world.
> > Just an option we can think about the method to access just the whole state of Target instance, something like
> > ```ts
> >    this.hasListeners('change'); // or similar short name
> > ```

> >   
> > From the usage point of view as you can see from the occurrence in the current PR it's more than enough to cover needs of `SyntheticEventTarget`
> > So, in case you prefer to enclose observers and go with the conservative `has` method instead of `get` mark this comment with the like +1emojy
> 
> this.hasListeners won't brake any of the code that is using SyntheticEventTarget api so I agree to change the method

According to the discussion results
We don't gonna share inner observers, but just let users know if they are presented.
So yea, @fshovchko, let's do this update

File: src/modules/esl-utils/dom/events/target.ts [hunk 1]
```diff
@@ -4,7 +4,12 @@
  * Ignores event type
  */
 export class SyntheticEventTarget implements EventTarget {
-  protected readonly _listeners = new Set<EventListenerOrEventListenerObject>();
+  private readonly _listeners = new Set<EventListenerOrEventListenerObject>();
+
+  // Parameter "type" will be used in future update
+  public getEventListenersLength(type: string): number {
```
Comment (ala-n) [inline src/modules/esl-utils/dom/events/target.ts:10]:
```ts
  public hasEventListener(type: string, minCount: number = 0): boolean;
```

**Review by dshovchko — APPROVED** [review]

**Review by abarmina — APPROVED** [review]

**Review by NastaLeo — APPROVED** [review]

**Review by ala-n — APPROVED** [review]

**Pull request merged**